src/basilisp/lang/compiler/analyzer.py

```diff
@@ -1432,7 +1436,8 @@ def __fn_method_ast(  # pylint: disable=too-many-branches,too-many-locals
 
         fn_loop_id = genname("fn_arity" if fnname is None else fnname.name)
         with ctx.new_recur_point(fn_loop_id, param_nodes):
-            stmts, ret = _body_ast(ctx, form.rest)
+            with ctx.expr_pos():
```

Comment:
I'm unaware of how to actually force this to happen for function nodes, but this felt like a safe transformation.